File: Sherlock and the Beast/SherlockAndTheBeast.py
'''
Created on 29/07/2014

@author: miguel
'''

if __name__ == '__main__':
    
    def sherlock(n):
        digits = n;
        threes = 0
        fives = 0
        while(digits > 0):
            if(digits % 3 == 0):
                fives = digits
                break
            digits -= 5
        if(digits < 0 or threes % 5 != 0):
            return -1
        threes = n-digits;
        fiveList = [5] * fives
        threeList = [3] * threes
        return "".join(str(x) for x in (fiveList + threeList))
                
    nCases = int(input())
    for i in range(nCases):
        n = int(input())
        print(sherlock(n))

File: Sherlock and the Beast/SherlockAndTheBeast.java
import java.util.Scanner;


/*
 * Sample Input
	
	4
	1
	3
	5
	11
	Sample Output
	
	-1
	555
	33333
	55555533333
 */


public class SherlockAndTheBeast {

	static String sherlock(int n){
		int digits = n;
		int fives = 0;
		int threes = 0;
		
		while(digits > 0){
			if(digits % 3 == 0){
				fives = digits;
				break;
			}
		digits -= 5;
		}
		threes = n - digits;
		if(threes % 5 != 0 || digits < 0)
			return "-1";
		StringBuilder answer = new StringBuilder();
		while(fives-- > 0)
			answer.append(5);
		while(threes-- > 0)
			answer.append(3);
		return answer.toString();
	}
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int nCases = sc.nextInt();
		for(int i = 0; i < nCases; i++){
			int n = sc.nextInt();
			System.out.println(sherlock(n));
		}
		sc.close();
	}
}
